pkg/smokescreen/smokescreen.go

```diff
@@ -519,12 +557,12 @@ func BuildProxy(config *Config) *goproxy.ProxyHttpServer {
 		defer logProxy(config, pctx)
 		defer pctx.Req.Header.Del(traceHeader)
 
-		err := handleConnect(config, pctx)
+		destination, err := handleConnect(config, pctx)
 		if err != nil {
 			pctx.Resp = rejectResponse(pctx, err)
 			return goproxy.RejectConnect, ""
 		}
-		return goproxy.OkConnect, host
+		return goproxy.OkConnect, destination
```

Comment:
this is kind of interesting: we're now using the normalized host name instead of the one passed on the CONNECT request (which varies based on http clients). I think this makes sense though!

Reply:
We should use the same host that we used for ACL checking. This way, even if someone is able to workaround host normalization, they only should be able to access destinations they were already authorized for.